From fbdac1339637e058795e0ce83b3c16e8c4ec8939 Mon Sep 17 00:00:00 2001
From: Luca Bacci <luca.bacci982@gmail.com>
Date: Sat, 15 Apr 2023 19:36:32 +0200
Subject: [PATCH] Fix check for Cairo DWrite Font dependency

Name changed from 'cairo-win32-dwrite-font' to 'cairo-dwrite-font',

https://gitlab.freedesktop.org/cairo/cairo/-/commit/15855876685f01af828c29ca2a1608eb06de2a89
---
 meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index fcd0bee8f..930f41082 100644
--- a/meson.build
+++ b/meson.build
@@ -363,7 +363,11 @@ cairo_dwrite_dep = disabler() # Only for Windows
 if cairo_dep.found()
   cairo_found_type = cairo_dep.type_name()
   if host_system == 'windows'
-    cairo_dwrite_dep = dependency('cairo-win32-dwrite-font', version: cairo_req_version, required: false)
+    cairo_dwrite_dep = dependency('cairo-dwrite-font', version: cairo_req_version, required: false)
+    # cairo < 1.17.8 used a different name
+    if not cairo_dwrite_dep.found()
+      cairo_dwrite_dep = dependency('cairo-win32-dwrite-font', version: cairo_req_version, required: false)
+    endif
   endif
 else
   if cc.get_id() == 'msvc' and cc.has_header('cairo.h')
-- 
GitLab

